quaireaux/searching/src/tests/binary_search_test.cairo

```diff
@@ -0,0 +1,89 @@
+use quaireaux_searching::binary_search::binary_search;
+use array::ArrayTrait;
+use option::OptionTrait;
+
+#[test]
+#[available_gas(2000000)]
+fn value_found() {
+    let mut arr: Array<u128> = ArrayTrait::new();
+
+    arr.append(100); // 0
+    arr.append(200); // 1
+    arr.append(300); // 2
+    arr.append(400); // 3
+    arr.append(500); // 4
+    arr.append(600); // 5
+
+    let mut span = arr.span();
+
+    // Test with an even length
+    assert(binary_search(span, 100).unwrap() == 0, 'Incorrect index');
```

Comment:
Could you please add something on the error message to identify more easily which line is failing, such as 
`Should be index 0`